pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+/* Begin Create Definition */
+
+type brokerCreateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	kapi.NameGenerator  // GenerateName method for CreateStrategy
+}
+
+// Strategy implements the call backs for the generic store
+var createStrategy = brokerCreateStrategy{
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	kapi.Scheme,
+	kapi.SimpleNameGenerator,
+}
+
+func (brokerCreateStrategy) Canonicalize(obj runtime.Object) {}
+
+// Are brokers namespace scoped?
+func (brokerCreateStrategy) NamespaceScoped() bool {
+	return false
+}
+
+func (brokerCreateStrategy) PrepareForCreate(ctx kapi.Context, obj runtime.Object) {
+	_ = obj.(*servicecatalog.Broker)
+}
+
+func (brokerCreateStrategy) Validate(ctx kapi.Context, obj runtime.Object) field.ErrorList {
+	return validateBroker(obj.(*servicecatalog.Broker))
+}
+
+/* End Create Definition */
+
+/* Begin Delete Definition */
+
+type brokerDeleteStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+}
+
+// Strategy implements the call backs for the generic store
+var deleteStrategy = brokerDeleteStrategy{
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	kapi.Scheme,
+}
+
+type brokerUpdateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+}
+
+// There is no implementation code for Delete.
+/* End Delete Definition */
+
+/* Begin Update Definition */
+
+// Strategy implements the call backs for the generic store
+var updateStrategy = brokerUpdateStrategy{
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	kapi.Scheme,
+}
+
+func (brokerUpdateStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (brokerUpdateStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (brokerUpdateStrategy) Canonicalize(obj runtime.Object) {}
```

Comment:
Canonicalize is called after Validate. What happens if it creates an object to persist that does not pass validate? (I think the answer is "don't do that")

I see a lot of empty methods as implementations.